src/adapters/DebugProtocolAdapter.ts

```diff
@@ -718,7 +717,21 @@ export class DebugProtocolAdapter {
         }
         return this.resolve('threads', async () => {
             let threads: Thread[] = [];
-            let threadsResponse = await this.client.threads();
+            let threadsResponse: ThreadsResponse;
+            for (let i = 0; i < 3; i++) {
+                threadsResponse = await this.client.threads();
+                if (threadsResponse.data.errorCode === ErrorCode.NOT_STOPPED) {
+                    this.logger.log(`Threads request retrying... ${i}:\n`, threadsResponse);
+                    threadsResponse = undefined;
+                    const pauseResponse = await this.client.pause(true);
+                    await util.sleep(100);
+                } else {
+                    break;
+                }
+            }
+            if (!threadsResponse) {
+                throw new Error('Failed to get threads data');
```

Comment:
We shouldn't throw an exception here because that might bring down the process. Instead, we should return some fake/dummy threads array that makes it apparent to the user that something is wrong. You might need to play with this a bit to get the structure right.

Reply:
I'll toy around with that in an hour or so. Thanks.